kymatio/scattering1d/backend/numpy_backend.py

```diff
@@ -0,0 +1,214 @@
+# Authors: Edouard Oyallon, Joakim Anden, Mathieu Andreux
+from collections import namedtuple
+import numpy as np
+
+BACKEND_NAME = 'numpy'
+
+def modulus_complex(x):
+    """Compute the complex modulus
+
+    Computes the modulus of x and stores the result in a complex tensor of the
+    same size, with the real part equal to the modulus and the imaginary part
+    equal to zero.
+
+    Parameters
+    ----------
+    x : tensor
+        A complex tensor (that is, whose last dimension is equal to 2).
+
+    Returns
+    -------
+    norm : tensor
+        A tensor with the same dimensions as x, such that norm[..., 0] contains
+        the complex modulus of x, while norm[..., 1] = 0.
+    """
+    
+    norm = np.abs(x)
+
+    return norm
+
+def subsample_fourier(x, k):
+    """Subsampling in the Fourier domain
+
+    Subsampling in the temporal domain amounts to periodization in the Fourier
+    domain, so the input is periodized according to the subsampling factor.
+
+    Parameters
+    ----------
+    x : tensor
+        Input tensor with at least 3 dimensions, where the next to last
+        corresponds to the frequency index in the standard PyTorch FFT
+        ordering. The length of this dimension should be a power of 2 to
+        avoid errors. The last dimension should represent the real and
+        imaginary parts of the Fourier transform.
+    k : int
+        The subsampling factor.
+
+    Returns
+    -------
+    res : tensor
+        The input tensor periodized along the next to last axis to yield a
+        tensor of size x.shape[-2] // k along that dimension.
+    """
+
+    N = x.shape[-2]
+    res = x.view(x.shape[:-2] + (k, N // k )).mean(dim=-1)
```

Comment:
Um... does `ndarray.view` work this way? Not according to [the docs](https://docs.scipy.org/doc/numpy/reference/generated/numpy.ndarray.view.html).